qrenderdoc/Windows/PipelineState/VulkanPipelineStateViewer.cpp

```diff
@@ -946,7 +946,7 @@ void VulkanPipelineStateViewer::addResourceRow(ShaderReflection *shaderDetails,
       }
       else if(bindType == BindType::Sampler)
       {
-        if(descriptorBind == NULL || descriptorBind->sampler == ResourceId())
+        if(descriptorBind != NULL || descriptorBind->sampler == ResourceId())
```

Comment:
I don't understand this change. The intention was that if the descriptor is NULL, or the descriptor doesn't contain a sampler, then the node is created with empty contents. Operator shortcutting means that if `descriptorBind` is NULL it won't dereference the pointer for the second check. What was the static analysis warning here exactly?

Reply:
You're right. That's completely wrong.

Reply:
`QString name =  descriptorBind->res;` on line 1208 is unchecked. It's not immediately obvious to me what name should be in that case, so I'll leave that for you to address.

Reply:
Ah yes I see, that will break. I'll fix that - it just needs to be a default `ResourceId()` instead.